tests/unit/test_collector.py

```diff
@@ -417,6 +398,66 @@ def test_ensure_quoted_url(url: str, clean_url: str) -> None:
     assert _ensure_quoted_url(url) == clean_url
 
 
+# versions containing fix/backport from https://github.com/python/cpython/pull/113563
+has_new_urlun_behaviour = (
+    # https://github.com/python/cpython/commit/387ff96e95b9f8a8cc7e646523ba3175b1350669
+    (sys.version_info[:2] == (3, 12) and sys.version_info >= (3, 12, 4))
```

Comment:
if we only support the latest version of Python in development we could probably drop the second check here, i.e. just check Python is `3.12` or `3.13` and inform anyone running into failures to use a newer Python. But maybe it's more convenient to just maintain the specific version for just a short time